h2o-algos/src/main/java/hex/glm/GLMModel.java

```diff
@@ -1161,14 +1178,31 @@ public GLMOutput(DataInfo dinfo, String[] column_names, String[] column_types, S
 
     public GLMOutput(DataInfo dinfo, String[] column_names, String[] column_types, String[][] domains, String[] coefficient_names, boolean binomial, double[] beta) {
       this(dinfo,column_names,column_types, domains,coefficient_names,binomial);
-      assert !ArrayUtils.hasNaNsOrInfs(beta);
+      assert !ArrayUtils.hasNaNsOrInfs(beta): "Coefficients contain NA or Infs.";
       _global_beta=beta;
       _submodels = new Submodel[]{new Submodel(0,beta,-1,Double.NaN,Double.NaN)};
     }
 
+    public GLMOutput(DataInfo dinfo, String[] column_names, String[] column_types, String[][] domains, 
```

Comment:
this is poor API design having one constructor like `... double[] beta, boolean isMultinomial` and another `boolean binomial, double[] beta` is very confusing, could we merge those two, maybe have two boolean params or better an Enum since both cannot be true at once I would think?

Reply:
I just combine all of them into one constructor.